client/mysqldump.c

```diff
@@ -5652,7 +5693,10 @@ static int start_transaction(MYSQL *mysql_con)
     if (!ignore_errors)
       exit(EX_MYSQLERR);
   }
-
+  if (has_session_variables_like(mysql_con, "rocksdb_skip_fill_cache"))
```

Comment:
Better to query this before any locking, i.e. move from start_transation to its own function, call it if (opt_single_transaction) from main